File: sdk/typescript/src/___PLAYGROUND___/signing/sign-text.ts
import { begin } from '../helpers';

export const signText = async () => {
  const context = await begin();

  console.log('Signing text message...');
  const message = `Hello from SDK test at ${new Date().toISOString()}`;
  const signature = await context.relayer.sign.text(message);

  console.log('Message:', message);
  console.log('Signature:', signature);

  await context.end();
};

signText().then(() => console.log('sign-text done'));

File: sdk/python/src/__PLAYGROUND__/signing/sign-text.py
import asyncio
from datetime import datetime

from __PLAYGROUND__.helper import begin, end


async def signText():
    rrelayer_node = None
    try:
        _, relayer, _, rrelayer_node = await begin()

        print("Signing text message...")

        message = f"Hello from SDK test at {datetime.now().isoformat()}"
        signature = await relayer.sign.text(message)
        print("Message:", message)
        print("Signature:", signature)
    except Exception as e:
        print("signText failed:", e)
    finally:
        if rrelayer_node is not None:
            end(rrelayer_node)


if __name__ == "__main__":
    asyncio.run(signText())
    print("sign-text done")
